### PR TITLE
Upgrade apollo client

### PR DESCRIPTION
**Pull request by tahmidrahman-dsi**

### CODE REVIEW

File: packages/client/src/utils/constants.ts [hunk 1]
```diff
@@ -94,10 +94,10 @@ export const MONTHS_IN_YEAR = 12
 
 /* change to import.meta.env.mode when migrating to vitest */
 export const LOADER_MIN_DISPLAY_TIME =
-  process.env.NODE_ENV !== 'test' ? 3 * 1000 : 0 // 3 seconds except test environment
+  import.meta.env.NODE_ENV !== 'test' ? 3 * 1000 : 0 // 3 seconds except test environment
```
Comment (Zangetsu101) [inline packages/client/src/utils/constants.ts:97]: @tahmidrahman-dsi We probably want to use `env.MODE` here instead

File: packages/client/src/utils/constants.ts [hunk 2]
```diff
@@ -94,10 +94,10 @@ export const MONTHS_IN_YEAR = 12
...
 
 export const DECLARED_DECLARATION_SEARCH_QUERY_COUNT =
-  process.env.DECLARED_DECLARATION_SEARCH_QUERY_COUNT || 100
+  import.meta.env.DECLARED_DECLARATION_SEARCH_QUERY_COUNT || 100
```
Comment (Zangetsu101) [inline packages/client/src/utils/constants.ts:100]: We should add the typing for this variable here: https://github.com/opencrvs/opencrvs-core/blob/99748a9bc612f24a0b111e69d74bf5025fae70c8/packages/client/typings/env.d.ts#L16

**Review by Zangetsu101 — CHANGES_REQUESTED** [review]

File: packages/client/src/utils/constants.ts (hunk 1 above)
Comment (tahmidrahman-dsi) [reply packages/client/src/utils/constants.ts]: Thanks @Zangetsu101 , done in https://github.com/opencrvs/opencrvs-core/pull/4047/commits/1dce985488c2886aa2c1b08400e82cc30c2eceea

File: packages/client/src/utils/constants.ts (hunk 2 above)
Comment (tahmidrahman-dsi) [reply packages/client/src/utils/constants.ts:100]: Thanks @Zangetsu101 ,  done in https://github.com/opencrvs/opencrvs-core/pull/4047/commits/1dce985488c2886aa2c1b08400e82cc30c2eceea

**Pull request merged**